### DISCUSSION TITLE
Get Oqtane Logs

### DISCUSSION THREAD

**Discussion by pnicosia80** [oqtane/oqtane.framework#4379] [Q&A]
I am creating a dashboard that displays Oqtane logs generated by my job, to be filtered based on the level and feature. I have found this method, or in any case, I would like to query the database directly to extend it. How do I obtain the current dbContextFactory in the controller of my module?

![image](https://github.com/oqtane/oqtane.framework/assets/10197467/517721e8-81a7-4427-ad4b-5e347c9b8cf6)

**Comment by hishamco** [comment]
I think you can filter the data by category to get your log entries

https://github.com/oqtane/oqtane.framework/blob/0efb3c32840f73fb6aeed1feaa0dc6e356f97de6/Oqtane.Shared/Models/Log.cs#L49

**Reply by pnicosia80** [reply]
Yes, but how do I obtain the current dbContextFactory in the controller of my module?

**Reply by hishamco** [reply]
It's there in line 20

**Comment by sbwalker** [comment]
@pnicosia80 if you create a module using the Create Module feature in the framework:

- login as host user
- navigate to Control Panel - Admin Dashboard 
- choose Module Management
- click Create Module button
- enter fields and click Create

It will scaffold the code for a fully functional module - including an API Controller class. This will demonstrate how to use DbContextFactory.
